### PR TITLE
intial impl / example of using audit lib

### PR DESCRIPTION
**Pull request by daiLlew**
Initial changes to api to use new go-ns audit lib. Not finished by any means but should give an overview of how to use the audit lib.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -107,6 +148,12 @@ func (api *ImportAPI) getJobs(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	setJSONContentType(w)
+
+	if err := api.auditor.Record(r.Context(), getJobsAuditAction, actionSuccessful, nil); err != nil {
+		log.ErrorC("error while attempting to audit event, failing request", err, nil)
```
Comment (eldeal) [inline api/api.go]: Consider making this 'audit failed' error event string a common string and reusing it throughout - the one above for getJobs and below for getJob are all slightly different so would be difficult to consistently find all audit failures in log aggregation. the action should be in the error already (or better yet we could add it to log.Data)

File: api/api.go [hunk 2]
```diff
@@ -132,50 +148,145 @@ func (api *ImportAPI) getJob(w http.ResponseWriter, r *http.Request) {
...
+	}
+
 	_, err = w.Write(bytes)
 	if err != nil {
 		log.Error(err, log.Data{})
 		http.Error(w, internalError, http.StatusInternalServerError)
 		return
 	}
-	log.Info("Returning an import job", log.Data{"id": jobID})
+	log.Info("Returning an import job", log.Data{jobIDKey: jobID})
```
Comment (mattrout92) [inline api/api.go]: may as well just use your logData variable here

File: api/api.go [hunk 3]
```diff
@@ -132,50 +148,145 @@ func (api *ImportAPI) getJob(w http.ResponseWriter, r *http.Request) {
...
+}
+
+func (api *ImportAPI) addJob(w http.ResponseWriter, r *http.Request) {
+	job, err := models.CreateJob(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.Error(err, log.Data{})
+		http.Error(w, "Bad client request received", http.StatusBadRequest)
```
Comment (mattrout92) [inline api/api.go]: is it worth adding to the handleAuditError for consistency?

File: cmd/dp-import-api/main.go [hunk 4]
```diff
@@ -127,6 +142,9 @@ func main() {
 	case err := <-directProducer.Errors():
 		log.ErrorC("kafka direct producer", err, nil)
 		shutdownGracefully(false)
+		/*	case err := <-auditProducer.Errors():
```
Comment (mattrout92) [inline cmd/dp-import-api/main.go]: Can this be uncommented now?

File: cmd/dp-import-api/main.go [hunk 5]
```diff
@@ -56,12 +60,22 @@ func main() {
...
+	auditProducer, err := kafka.NewProducer(config.Brokers, "audit-events", config.KafkaMaxBytes)
+	if err != nil {
+		log.ErrorC("direct kafka producer error", err, nil)
+		os.Exit(1)
+	}
 
 	router := mux.NewRouter()
 	router.Path("/healthcheck").HandlerFunc(healthcheck.Handler)
 
+	// TODO replace with impl when ready.
```
Comment (mattrout92) [inline cmd/dp-import-api/main.go]: can this be done now?

File: config/config.go [hunk 6]
```diff
@@ -55,13 +56,16 @@ func Get() (*Configuration, error) {
...
 		ZebedeeURL:              "http://localhost:8082",
+		AuditEventTopic:         "audit-events",
 	}
 
 	err := envconfig.Process("", cfg)
 	if err != nil {
 		return nil, err
 	}
 
+	//cfg.ServiceAuthToken = "Bearer " + cfg.ServiceAuthToken
```
Comment (mattrout92) [inline config/config.go]: Should this be deleted

File: config/config_test.go [hunk 7]
```diff
@@ -13,8 +13,9 @@ func TestGetReturnsDefaultValues(t *testing.T) {
 		So(configuration.BindAddr, ShouldEqual, ":21800")
 		So(configuration.DatabakerImportTopic, ShouldEqual, "data-bake-job-available")
 		So(configuration.SecretKey, ShouldEqual, "0C30662F-6CF6-43B0-A96A-954772267FF5")
-		So(configuration.ServiceAuthToken, ShouldEqual, "0C30662F-6CF6-43B0-A96A-954772267FF5")
+		So(configuration.ServiceAuthToken, ShouldEqual, "Bearer 0C30662F-6CF6-43B0-A96A-954772267FF5")
```
Comment (mattrout92) [inline config/config_test.go]: I don't think Bearer should be included here, it should be being set by the client

Comment (daiLlew) [reply config/config_test.go]: good spot

**Pull request merged**